lib/cluster-providers/types.ts

```diff
@@ -5,7 +5,7 @@ import { AutoScalingGroupCapacityOptions } from "aws-cdk-lib/aws-eks";
 /**
  * Configuration options for the custom AMI.
  */
-export interface CustomAmiProps {
+export interface LaunchTemplateProps {
```

Comment:
Do you mind checking if it is worth including anything else here for the launch template? I am thinking `spotOptions` and potentially `keyName`  for SSH.